src/Nethermind/Nethermind.Merge.Plugin/Data/ExecutionPayload.cs

```diff
@@ -215,4 +201,11 @@ public static class ExecutionPayloadExtensions
             specProvider.GetSpec(executionPayload.BlockNumber, executionPayload.Timestamp),
             version,
             out error);
+
+    public static bool IsProperFork(this ExecutionPayload executionPayload, ISpecProvider specProvider)
```

Comment:
If we are going class hierarchy route, maybe this should be a virtual method on `ExcecutionPayload` ?

Reply:
Agreed, moved to clas hierarchy